src/reason_toolchain.ml

```diff
@@ -487,9 +510,14 @@ module JS_syntax = struct
        loop_handle_yacc supplier in_error checkpoint
     | I.HandlingError env ->
        if !Reason_config.recoverable then
+         (
+         let loc = last_token_loc supplier in
+         let state = state checkpoint in
```

Comment:
It would be nice if we could call this without side-effects.

Without this: all the errors are visible, but not the nice Menhir ones.
With this: sometimes only shows 1 Menhir messages error